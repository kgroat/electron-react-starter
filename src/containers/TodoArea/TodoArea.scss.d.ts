export const todoArea: string;
export const helperBar: string;
export const title: string;
export const hideDone: string;
export const icon: string;
