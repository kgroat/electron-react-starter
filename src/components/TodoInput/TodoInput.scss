
@import '../../globalStyles/mixins';

$greyText: rgba($primaryDarkText, 0.5);

.todoInput {
  @include itemBox();
  display: flex;
  position: relative;

  .title {
    flex-grow: 1;
    z-index: 1;
    min-width: 0;
  }

  .titleDefault {
    padding: inherit;
    position: absolute;
    top: 0;
    left: 0;
    color: $greyText;
  }

  .submitButton {
    height: 0;
    background: none;
    border: none;
    color: inherit;
    font-size: inherit;
    outline: none;
    padding: 0;
    padding-left: 1.4em;
    margin-top: -0.5em;

    .icon, .icon.disabled {
      padding-left: 0;
    }
  }

  .icon {
    @include icon(inherit, $hoverColor);
    
    &.disabled {
      @include icon($greyText, $greyText);
    }
  }
}
