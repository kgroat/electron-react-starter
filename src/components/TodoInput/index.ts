
export { default, State, Props } from './TodoInput'
