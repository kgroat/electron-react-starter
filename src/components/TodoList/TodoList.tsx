
import * as React from 'react'

import { Todo } from 'state'
import TodoItem from 'components/TodoItem'

import { todoList } from './TodoList.scss'

export interface Props {
  items: Todo[]
  toggleTodo: (id: string, checked: boolean) => void
  deleteTodo: (id: string) => void
}

export default function TodoList ({ items, toggleTodo, deleteTodo }: Props) {
  return (
    <div className={todoList}>
      {
        items.map(item => (
          <TodoItem
            key={item.id}
            name={item.name}
            checked={item.checked}
            onChange={checked => toggleTodo(item.id, checked)}
            onDelete={() => deleteTodo(item.id)} />
        ))
      }
    </div>
  )
}
