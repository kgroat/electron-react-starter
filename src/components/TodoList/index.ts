
export { default, Props } from './TodoList'
