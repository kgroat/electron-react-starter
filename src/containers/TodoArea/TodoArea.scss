
@import '../../globalStyles/mixins';

.todoArea {
  .helperBar {
    @include itemBox();
    background: $secondaryLight;
    color: $secondaryText;
    padding-top: 0.1em;
    padding-bottom: 0.1em;
    display: flex;

    .title {
      flex-grow: 1;
    }

    .hideDone {
      flex-shrink: 1;
      
      .icon {
        @include icon($secondaryText, $secondaryHover);
        font-size: 1em;
      }
    }
  }
}
