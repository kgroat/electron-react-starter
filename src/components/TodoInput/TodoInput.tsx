
import * as React from 'react'

import { todoInput, title, titleDefault, submitButton, icon, disabled } from './TodoInput.scss'

export interface State {
  name: string
  checked: boolean
}

export interface Props {
  onAdd: (name: string, checked: boolean) => void
}

export default class TodoInput extends React.Component<Props, State> {
  state = {
    name: '',
    checked: false
  }

  setName = (name: string) =>
    this.setState({ name })

  toggleChecked = (checked = !this.state.checked) =>
    this.setState({ checked })

  onAdd = (ev) => {
    ev.preventDefault()
    if (!this.state.name) {
      return
    }
    this.props.onAdd(this.state.name, this.state.checked)
    this.setState({ name: '', checked: false })
  }

  onNameChange = (ev: React.ChangeEvent<HTMLInputElement>) =>
    this.setName(ev.currentTarget.value)

  render () {
    const checkboxIcon = this.state.checked ? 'checkbox-marked-outline' : 'checkbox-blank-outline'

    return (
      <form className={todoInput} onSubmit={this.onAdd}>
        <input className={title} onChange={this.onNameChange} value={this.state.name} />
        { !this.state.name ? <span className={titleDefault}>Choose Name</span> : '' }
        <i onClick={() => this.toggleChecked()} className={`checkbox mdi mdi-${checkboxIcon} ${icon}`} />
        <button type='submit' className={submitButton}>
          <i className={`mdi mdi-plus-box ${icon} ${!this.state.name ? disabled : ''}`} />
        </button>
      </form>
    )
  }
}
