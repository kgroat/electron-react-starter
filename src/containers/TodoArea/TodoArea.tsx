
import * as React from 'react'
import { connect, Dispatch } from 'react-redux'

import { State, Todo } from 'state'
import { addTodo, toggleTodo, deleteTodo } from 'actions'

import TodoInput from 'components/TodoInput'
import TodoList from 'components/TodoList'

import { todoArea, helperBar, title, hideDone as hideDoneClass, icon } from './TodoArea.scss'

interface StateProps {
  todos: Todo[]
}

interface DispatchProps {
  addTodo: (name: string, checked: boolean) => void
  toggleTodo: (id: string, checked: boolean) => void
  deleteTodo: (id: string) => void
}

export type Props = StateProps & DispatchProps

interface LocalState {
  hideDone: boolean
}

export class TodoAreaBase extends React.Component<Props, LocalState> {
  state = {
    hideDone: true
  }

  toggleHideDone = (hideDone = !this.state.hideDone) =>
    this.setState({ hideDone })

  render () {
    const { hideDone } = this.state

    const todos = hideDone
      ? this.props.todos.filter(item => !item.checked)
      : this.props.todos

    const checkboxIcon = hideDone ? 'checkbox-marked-outline' : 'checkbox-blank-outline'

    return (
      <div className={todoArea}>
        <TodoInput onAdd={this.props.addTodo} />
        <div className={helperBar}>
          <span className={title}>
            Title
          </span>
          <span className={hideDoneClass}>
            Hide done?
            <i className={`mdi mdi-${checkboxIcon} ${icon}`} onClick={() => this.toggleHideDone()} />
          </span>
        </div>
        <TodoList items={todos} toggleTodo={this.props.toggleTodo} deleteTodo={this.props.deleteTodo} />
      </div>
    )
  }
}

function mapStateToProps ({ todos }: State): StateProps {
  return {
    todos
  }
}

function mapDispatchToProps (dispatch: Dispatch<State>): DispatchProps {
  return {
    addTodo: (name, checked) => dispatch(addTodo({ name, checked })),
    toggleTodo: (id, checked) => dispatch(toggleTodo({ id, checked })),
    deleteTodo: (id) => dispatch(deleteTodo({ id }))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(TodoAreaBase)
