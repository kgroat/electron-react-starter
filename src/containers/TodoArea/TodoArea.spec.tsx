
import * as React from 'react'
import { create } from 'react-test-renderer'
import { Provider } from 'react-redux'
import { Store, Dispatch } from 'redux'
import { shallow } from 'enzyme'

import { State, Todo } from 'state'
import TodoInput from 'components/TodoInput'
import TodoList from 'components/TodoList'

import {
  addTodoType, AddTodoAction,
  toggleTodoType, ToggleTodoAction,
  deleteTodoType, DeleteTodoAction
} from 'actions'

import { TodoAreaBase, Props } from './TodoArea'
import { icon } from './TodoArea.scss'
import TodoArea from './'

let state: State

let dispatch: Dispatch<State> & jest.Mock<any>

const store = {
  getState () {
    return state
  },
  get dispatch () {
    return dispatch as Dispatch<State>
  },
  subscribe: jest.fn(),
  replaceReducer: jest.fn()
} as Store<State>

function resetState () {
  state = {
    todos: []
  } as State
}

function createTodo (id: string, checked: boolean): Todo {
  return {
    id,
    name: 'Get milk',
    checked
  }
}

function createTodos (count: number, checked): Todo[] {
  return Array.from(Array(count))
    .map((v, idx) => createTodo(`${idx}`, checked))
}

const defaultProps: Props = {
  todos: [],
  addTodo: jest.fn(),
  toggleTodo: jest.fn(),
  deleteTodo: jest.fn()
}

describe('<TodoArea />', () => {
  beforeEach(() => {
    resetState()
    dispatch = jest.fn()
  })

  describe('snapshots', () => {
    [1, 2, 3, 4, 5].forEach(i => {
      it(`should render with ${i} unchecked todo items`, () => {
        state.todos = createTodos(i, false)
        const tree = create(
          <Provider store={store}>
            <TodoArea />
          </Provider>
        )
        expect(tree).toMatchSnapshot()
      })

      it(`should render with ${i} checked todo items`, () => {
        state.todos = createTodos(i, true)
        const tree = create(
          <Provider store={store}>
            <TodoArea />
          </Provider>
        )
        expect(tree).toMatchSnapshot()
      })
    })

    it('should render without any todo items', () => {
      const tree = create(
        <Provider store={store}>
          <TodoArea />
        </Provider>
      )
      expect(tree).toMatchSnapshot()
    })
  })

  describe('user adds a todo item', () => {
    it(`should call addTodo once`, () => {
      const addTodo = jest.fn()
      const name = 'Get milk'

      const element = shallow(<TodoAreaBase {...defaultProps} addTodo={addTodo} />)

      element.find(TodoInput).props().onAdd(name, false)

      expect(addTodo).toHaveBeenCalledWith(name, false)
      expect(addTodo).toHaveBeenCalledTimes(1)
    })

    it(`should dispatch an ${addTodoType} action once`, () => {
      const name = 'Get milk'
      const expected: AddTodoAction = {
        type: addTodoType,
        name,
        checked: false
      }

      const element = shallow(<TodoArea />, { context: { store } })

      element.find(TodoAreaBase).props().addTodo(name, false)

      expect(dispatch).toHaveBeenCalledTimes(1)
      const actual = dispatch.mock.calls[0][0]
      expect(actual).toEqual(expected)
    })
  })

  describe('user checks or unchecks a todo item', () => {
    ['id1', 'id2', 'id3'].map(id => {
      [true, false].map(checked => {
        it(`should call toggleTodo with ${id} and ${checked} once`, () => {
          const toggleTodo = jest.fn()

          const element = shallow(<TodoAreaBase {...defaultProps} toggleTodo={toggleTodo} />)

          element.find(TodoList).props().toggleTodo(id, checked)

          expect(toggleTodo).toBeCalledWith(id, checked)
          expect(toggleTodo).toHaveBeenCalledTimes(1)
        })

        it(`should dispatch a ${toggleTodoType} action with id: ${id} and checked: ${checked} once`, () => {
          state.todos = [createTodo(id, true)]
          const expected: ToggleTodoAction = {
            type: toggleTodoType,
            id,
            checked
          }

          const element = shallow(<TodoArea />, { context: { store } })

          element.find(TodoAreaBase).props().toggleTodo(expected.id, expected.checked)

          expect(dispatch).toHaveBeenCalledTimes(1)
          const actual = dispatch.mock.calls[0][0]
          expect(actual).toEqual(expected)
        })
      })
    })
  })

  describe('user deletes a todo item', () => {
    ['id1', 'id2', 'id3'].map(id => {
      it(`should call deleteTodo with ${id} once`, () => {
        const deleteTodo = jest.fn()

        const element = shallow(<TodoAreaBase {...defaultProps} deleteTodo={deleteTodo} />)

        element.find(TodoList).props().deleteTodo(id)

        expect(deleteTodo).toHaveBeenCalledWith(id)
        expect(deleteTodo).toHaveBeenCalledTimes(1)
      })

      it(`should dispatch a ${deleteTodoType} action with id ${id} once`, () => {
        state.todos = [createTodo(id, false)]
        const expected: DeleteTodoAction = {
          type: deleteTodoType,
          id
        }

        const element = shallow(<TodoArea />, { context: { store } })

        element.find(TodoAreaBase).props().deleteTodo(id)

        expect(dispatch).toHaveBeenCalledTimes(1)
        const actual = dispatch.mock.calls[0][0]
        expect(actual).toEqual(expected)
      })
    })
  })

  describe('user shows done todos', () => {
    [1, 2, 3, 4, 5].forEach(i => {
      it(`should show ${i} todos if the checkbox remains checked if all are unchecked`, () => {
        const expected = i
        const todos = createTodos(expected, false)
        const element = shallow(<TodoAreaBase {...defaultProps} todos={todos} />)

        const actual = element.find(TodoList).props().items.length
        expect(actual).toBe(expected)
      })

      it(`should 0 todos if all are checked and the checkbox remains checked`, () => {
        const expected = 0
        const todos = createTodos(i, true)
        const element = shallow(<TodoAreaBase {...defaultProps} todos={todos} />)

        const actual = element.find(TodoList).props().items.length
        expect(actual).toBe(expected)
      })

      it(`should show ${i} todos if the checkbox is unchecked even if all are checked`, () => {
        const expected = i
        const todos = createTodos(i, true)
        const element = shallow(<TodoAreaBase {...defaultProps} todos={todos} />)

        element.findWhere(el => el.hasClass(icon)).simulate('click')

        const actual = element.find(TodoList).props().items.length
        expect(actual).toBe(expected)
      })
    })
  })
})
