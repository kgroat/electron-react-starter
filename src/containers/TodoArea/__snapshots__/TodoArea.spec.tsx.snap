// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoArea /> snapshots should render with 1 checked todo items 1`] = `
<div
  className="todoArea"
>
  <form
    className="todoInput"
    onSubmit={[Function]}
  >
    <input
      className="title"
      onChange={[Function]}
      value=""
    />
    <span
      className="titleDefault"
    >
      Choose Name
    </span>
    <i
      className="checkbox mdi mdi-checkbox-blank-outline icon"
      onClick={[Function]}
    />
    <button
      className="submitButton"
      type="submit"
    >
      <i
        className="mdi mdi-plus-box icon disabled"
      />
    </button>
  </form>
  <div
    className="helperBar"
  >
    <span
      className="title"
    >
      Title
    </span>
    <span
      className="hideDone"
    >
      Hide done?
      <i
        className="mdi mdi-checkbox-marked-outline icon"
        onClick={[Function]}
      />
    </span>
  </div>
  <div
    className="todoList"
  />
</div>
`;

exports[`<TodoArea /> snapshots should render with 1 unchecked todo items 1`] = `
<div
  className="todoArea"
>
  <form
    className="todoInput"
    onSubmit={[Function]}
  >
    <input
      className="title"
      onChange={[Function]}
      value=""
    />
    <span
      className="titleDefault"
    >
      Choose Name
    </span>
    <i
      className="checkbox mdi mdi-checkbox-blank-outline icon"
      onClick={[Function]}
    />
    <button
      className="submitButton"
      type="submit"
    >
      <i
        className="mdi mdi-plus-box icon disabled"
      />
    </button>
  </form>
  <div
    className="helperBar"
  >
    <span
      className="title"
    >
      Title
    </span>
    <span
      className="hideDone"
    >
      Hide done?
      <i
        className="mdi mdi-checkbox-marked-outline icon"
        onClick={[Function]}
      />
    </span>
  </div>
  <div
    className="todoList"
  >
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`<TodoArea /> snapshots should render with 2 checked todo items 1`] = `
<div
  className="todoArea"
>
  <form
    className="todoInput"
    onSubmit={[Function]}
  >
    <input
      className="title"
      onChange={[Function]}
      value=""
    />
    <span
      className="titleDefault"
    >
      Choose Name
    </span>
    <i
      className="checkbox mdi mdi-checkbox-blank-outline icon"
      onClick={[Function]}
    />
    <button
      className="submitButton"
      type="submit"
    >
      <i
        className="mdi mdi-plus-box icon disabled"
      />
    </button>
  </form>
  <div
    className="helperBar"
  >
    <span
      className="title"
    >
      Title
    </span>
    <span
      className="hideDone"
    >
      Hide done?
      <i
        className="mdi mdi-checkbox-marked-outline icon"
        onClick={[Function]}
      />
    </span>
  </div>
  <div
    className="todoList"
  />
</div>
`;

exports[`<TodoArea /> snapshots should render with 2 unchecked todo items 1`] = `
<div
  className="todoArea"
>
  <form
    className="todoInput"
    onSubmit={[Function]}
  >
    <input
      className="title"
      onChange={[Function]}
      value=""
    />
    <span
      className="titleDefault"
    >
      Choose Name
    </span>
    <i
      className="checkbox mdi mdi-checkbox-blank-outline icon"
      onClick={[Function]}
    />
    <button
      className="submitButton"
      type="submit"
    >
      <i
        className="mdi mdi-plus-box icon disabled"
      />
    </button>
  </form>
  <div
    className="helperBar"
  >
    <span
      className="title"
    >
      Title
    </span>
    <span
      className="hideDone"
    >
      Hide done?
      <i
        className="mdi mdi-checkbox-marked-outline icon"
        onClick={[Function]}
      />
    </span>
  </div>
  <div
    className="todoList"
  >
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`<TodoArea /> snapshots should render with 3 checked todo items 1`] = `
<div
  className="todoArea"
>
  <form
    className="todoInput"
    onSubmit={[Function]}
  >
    <input
      className="title"
      onChange={[Function]}
      value=""
    />
    <span
      className="titleDefault"
    >
      Choose Name
    </span>
    <i
      className="checkbox mdi mdi-checkbox-blank-outline icon"
      onClick={[Function]}
    />
    <button
      className="submitButton"
      type="submit"
    >
      <i
        className="mdi mdi-plus-box icon disabled"
      />
    </button>
  </form>
  <div
    className="helperBar"
  >
    <span
      className="title"
    >
      Title
    </span>
    <span
      className="hideDone"
    >
      Hide done?
      <i
        className="mdi mdi-checkbox-marked-outline icon"
        onClick={[Function]}
      />
    </span>
  </div>
  <div
    className="todoList"
  />
</div>
`;

exports[`<TodoArea /> snapshots should render with 3 unchecked todo items 1`] = `
<div
  className="todoArea"
>
  <form
    className="todoInput"
    onSubmit={[Function]}
  >
    <input
      className="title"
      onChange={[Function]}
      value=""
    />
    <span
      className="titleDefault"
    >
      Choose Name
    </span>
    <i
      className="checkbox mdi mdi-checkbox-blank-outline icon"
      onClick={[Function]}
    />
    <button
      className="submitButton"
      type="submit"
    >
      <i
        className="mdi mdi-plus-box icon disabled"
      />
    </button>
  </form>
  <div
    className="helperBar"
  >
    <span
      className="title"
    >
      Title
    </span>
    <span
      className="hideDone"
    >
      Hide done?
      <i
        className="mdi mdi-checkbox-marked-outline icon"
        onClick={[Function]}
      />
    </span>
  </div>
  <div
    className="todoList"
  >
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`<TodoArea /> snapshots should render with 4 checked todo items 1`] = `
<div
  className="todoArea"
>
  <form
    className="todoInput"
    onSubmit={[Function]}
  >
    <input
      className="title"
      onChange={[Function]}
      value=""
    />
    <span
      className="titleDefault"
    >
      Choose Name
    </span>
    <i
      className="checkbox mdi mdi-checkbox-blank-outline icon"
      onClick={[Function]}
    />
    <button
      className="submitButton"
      type="submit"
    >
      <i
        className="mdi mdi-plus-box icon disabled"
      />
    </button>
  </form>
  <div
    className="helperBar"
  >
    <span
      className="title"
    >
      Title
    </span>
    <span
      className="hideDone"
    >
      Hide done?
      <i
        className="mdi mdi-checkbox-marked-outline icon"
        onClick={[Function]}
      />
    </span>
  </div>
  <div
    className="todoList"
  />
</div>
`;

exports[`<TodoArea /> snapshots should render with 4 unchecked todo items 1`] = `
<div
  className="todoArea"
>
  <form
    className="todoInput"
    onSubmit={[Function]}
  >
    <input
      className="title"
      onChange={[Function]}
      value=""
    />
    <span
      className="titleDefault"
    >
      Choose Name
    </span>
    <i
      className="checkbox mdi mdi-checkbox-blank-outline icon"
      onClick={[Function]}
    />
    <button
      className="submitButton"
      type="submit"
    >
      <i
        className="mdi mdi-plus-box icon disabled"
      />
    </button>
  </form>
  <div
    className="helperBar"
  >
    <span
      className="title"
    >
      Title
    </span>
    <span
      className="hideDone"
    >
      Hide done?
      <i
        className="mdi mdi-checkbox-marked-outline icon"
        onClick={[Function]}
      />
    </span>
  </div>
  <div
    className="todoList"
  >
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`<TodoArea /> snapshots should render with 5 checked todo items 1`] = `
<div
  className="todoArea"
>
  <form
    className="todoInput"
    onSubmit={[Function]}
  >
    <input
      className="title"
      onChange={[Function]}
      value=""
    />
    <span
      className="titleDefault"
    >
      Choose Name
    </span>
    <i
      className="checkbox mdi mdi-checkbox-blank-outline icon"
      onClick={[Function]}
    />
    <button
      className="submitButton"
      type="submit"
    >
      <i
        className="mdi mdi-plus-box icon disabled"
      />
    </button>
  </form>
  <div
    className="helperBar"
  >
    <span
      className="title"
    >
      Title
    </span>
    <span
      className="hideDone"
    >
      Hide done?
      <i
        className="mdi mdi-checkbox-marked-outline icon"
        onClick={[Function]}
      />
    </span>
  </div>
  <div
    className="todoList"
  />
</div>
`;

exports[`<TodoArea /> snapshots should render with 5 unchecked todo items 1`] = `
<div
  className="todoArea"
>
  <form
    className="todoInput"
    onSubmit={[Function]}
  >
    <input
      className="title"
      onChange={[Function]}
      value=""
    />
    <span
      className="titleDefault"
    >
      Choose Name
    </span>
    <i
      className="checkbox mdi mdi-checkbox-blank-outline icon"
      onClick={[Function]}
    />
    <button
      className="submitButton"
      type="submit"
    >
      <i
        className="mdi mdi-plus-box icon disabled"
      />
    </button>
  </form>
  <div
    className="helperBar"
  >
    <span
      className="title"
    >
      Title
    </span>
    <span
      className="hideDone"
    >
      Hide done?
      <i
        className="mdi mdi-checkbox-marked-outline icon"
        onClick={[Function]}
      />
    </span>
  </div>
  <div
    className="todoList"
  >
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
    <div
      className="todoItem"
    >
      <span
        className="title"
      >
        Get milk
      </span>
      <i
        className="mdi mdi-delete deleteIcon"
        onClick={[Function]}
      />
      <i
        className="mdi mdi-checkbox-blank-outline checkboxIcon"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`<TodoArea /> snapshots should render without any todo items 1`] = `
<div
  className="todoArea"
>
  <form
    className="todoInput"
    onSubmit={[Function]}
  >
    <input
      className="title"
      onChange={[Function]}
      value=""
    />
    <span
      className="titleDefault"
    >
      Choose Name
    </span>
    <i
      className="checkbox mdi mdi-checkbox-blank-outline icon"
      onClick={[Function]}
    />
    <button
      className="submitButton"
      type="submit"
    >
      <i
        className="mdi mdi-plus-box icon disabled"
      />
    </button>
  </form>
  <div
    className="helperBar"
  >
    <span
      className="title"
    >
      Title
    </span>
    <span
      className="hideDone"
    >
      Hide done?
      <i
        className="mdi mdi-checkbox-marked-outline icon"
        onClick={[Function]}
      />
    </span>
  </div>
  <div
    className="todoList"
  />
</div>
`;
